packages/openapi3/src/openapi.ts

```diff
@@ -1307,6 +1308,16 @@ function createOAPIEmitter(program: Program, options: ResolvedOpenAPI3EmitterOpt
       newTarget.format = "password";
     }
 
+    const dateFormat = getDateFormat(program, typespecType);
```

Comment:
We should also set the default dateFormat, which requires knowing whether the date is used in a header  (rfc7231) or elsewhere (rfc 3339)

Reply:
We’ll in openapi date-time already assume that from what I understand.

Reply:
AFAIK, no, see: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md?plain=1#L170

date-time should be the default for non-header types, but for headers, we need to be explicit

Reply:
Oh I see, I’ll update